rest-api/src/test/java/gov/cms/qpp/conversion/api/services/StorageServiceImplTest.java

```diff
@@ -35,6 +21,23 @@
 import static org.mockito.Mockito.verify;
 import static org.powermock.api.mockito.PowerMockito.when;
 
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.Upload;
+import com.amazonaws.services.s3.transfer.model.UploadResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.core.env.Environment;
+import org.springframework.core.task.TaskExecutor;
+
 
 @ExtendWith(MockitoExtension.class)
 class StorageServiceImplTest {
```

Comment:
I would add another test to this file.  I would mock the `transferManager.upload(...)` call to first throw an exception and then on the invocation check to see that reading the `InputStream` returns something valid instead of an empty stream or some other exception that indicates that it has already been used.

Reply:
Wrote MeasuredInputStreamSupplierTest

Reply:
👍 Nice.